Makefile

```diff
@@ -1,9 +1,20 @@
 .DEFAULT_GOAL := package
 
-.PHONY: package
-package:
+.PHONY: check-install compile package
+package: compile
 	npm run package
 
 .PHONY: install
 install:
 	npm install
+
+.PHONY: check-install
+check-install:
+	@if [ ! -d "node_modules" ]; then \
+		echo "installing npm packages..."; \
+		npm install; \
+	fi
```

Comment:
Here's an efficient `make` target for installing npm packages:

```Makefile
node_modules: package-lock.json
    npm ci
```

It can be used as a pre-requisite, for example:
```Makefile
.PHONY: compile
compile: node_modules
	npm run compile
```

However, considering that this is a pure TS/JS project, shouldn't we simply remove the Makefile instead?

Reply:
Both `npm install` and `npm ci` will install - the relevant difference is that `npm ci` never modifies the lock file. 

With the make target above, installation will only happen once until the lock file changes, which is what you want.